pytensor/xtensor/rewriting/shape.py

```diff
@@ -105,3 +119,73 @@ def lower_transpose(fgraph, node):
     x_tensor_transposed = x_tensor.transpose(perm)
     new_out = xtensor_from_tensor(x_tensor_transposed, dims=out_dims)
     return [new_out]
+
+
+@register_xcanonicalize
+@node_rewriter([ExpandDims])
+def local_expand_dims_reshape(fgraph, node):
+    """Rewrite rule to convert expand_dims to pytensor.tensor.expand_dims and broadcast_to if needed."""
+    if not isinstance(node.op, ExpandDims):
+        return False
+
+    x = node.inputs[0]
+    dim = node.op.dim
+    size = getattr(node.op, "size", 1)
```

Comment:
size should be a symbolic input (or multiple if we have multiple dims) to the node, so you'll have `x, *sizes = node.inputs`. This way they can be arbitrary symbolic expressions and not just constants. Check how `unstack` does it.